Support for --static option via #67

abepark01 opened the pull request:
This pull request supports #67

dougwilson commented on README.md:
```diff
@@ -44,6 +44,7 @@ This generator can also be further configured with the following command line fl
     -V, --version       output the version number
     -e, --ejs           add ejs engine support (defaults to jade)
         --hbs           add handlebars engine support
+        --static        add support for static html support
```
duplicate word "support" makes it hard to read

nicolasmccurdy replied:
I think something like "use static html without templates" would be a more readable description. Also, can we add the `-s` shortcut?

abepark01 replied:
Fixed, just pushed a fix

dougwilson commented on templates/js/app-static.js:
```diff
@@ -0,0 +1,56 @@
+var express = require('express');
```
Do not duplicate this entire file for static support, because now we have to maintain two almost identical files. Please adjust the existing file to support both with template strings like we do already.

dougwilson commented on templates/js/routes/index-static.js:
```diff
@@ -0,0 +1,10 @@
+/* GET home page. */
+
+router.get('/', function(req, res, next) {
+  res.sendfile('public/index.html');
```
Do not duplicate this entire file for static support, because now we have to maintain two almost identical files. Please adjust the existing file to support both with template strings like we do already.

dougwilson commented:
Hi @abepark01 ! Are you interested in addressing the comments made on this PR?

abepark01 commented:
Yes! Sorry, I've been a bit busy, but would love to see how I can resolve.

I do understand concern about duplicating code, so trying to see what can be done.

Thanks!

Sent from my iPhone

> On Jul 3, 2015, at 8:21 PM, Douglas Christopher Wilson <notifications@github.com> wrote:
> 
> Hi @abepark01 ! Are you interested in addressing the comments made on this PR?
> 
> —
> Reply to this email directly or view it on GitHub.
>

nicolasmccurdy commented on package.json:
```diff
@@ -1,54 +1,17 @@
+    "serve-favicon": "~2.2.0"
   }
-}
+}
```
I don't think this file needs to be changed so much.

nicolasmccurdy commented on templates/js/app-static.js:
```diff
@@ -0,0 +1,56 @@
+app.use(logger('dev'));
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(cookieParser());{css}
```
Is `{css}` intentional?

nicolasmccurdy commented:
I added some comments, and I'm very interested in getting this feature implemented. Do you want me to make a pull request based off this?

abepark01 commented:
Hi guys!
Thanks for the feedback; I have updated the code base on the feedback provided.

Please review and let me know!

abepark01 commented:
Seems like the AppVeyor build is failing due to timeouts; any suggestions?

 1) express(1) (no args) should have installable dependencies:
     Error: timeout of 30000ms exceeded. Ensure the done() callback is being called in this test.

abepark01 commented:
Hi guys,

Please let me know if there are any additional issues you would like for me to address? I do believe this is ready to go out.

smithb1994 commented:
@abepark01  Sooooo.... any news on this getting merged?

abepark01 commented:
Hi @dougwilson!
Just wanted to see you got a chance to review the fixes I made in response to your feedback.
Thanks!
I would love to see this merged if everything checks out.

dougwilson commented on bin/express:
```diff
@@ -134,6 +142,21 @@ function createApplication(app_name, path) {
+      // set index route body
+      index = index.replace(
+        '{indexRouteBody}',
+        "res.sendfile('public/index.html');\n"
```
What is the purpose of having a route with sendFile? What logic would someone add to the route that cannot output anything? Wouldn't it be better to leave off the route all together and let express.static connect index.html to the / request? I believe this is what various Angular boilerplates do.

dougwilson commented on bin/express:
```diff
@@ -330,9 +353,11 @@ function main() {
   // Template engine
   program.template = 'jade';
   if (program.ejs) program.template = 'ejs';
+  if (program.static) program.template = 'ejs';
```
Why would a static site still have ejs in the package.json for?

dougwilson commented:
Please add tests for the static option to the test suite.

dougwilson commented:
Sorry this got closed from some GitHub voodoo. It doesn't have the option for me to re-open it form some reason! :( Please re-create your pull request.
